app/controllers/verify/finance_controller.rb

```diff
@@ -17,8 +18,25 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_form: idv_finance_form,
+        idv_session: idv_session,
+        analytics: analytics,
+        params: step_params
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
```

Comment:
What is the purpose of remembering whether or not a step has already been validated by the vendor? Is it to enforce that a user cannot submit a step again after it has been validated? Or is it as a convenience to the user so they don't have to see that step again?

Either way, shouldn't this be stored in the DB to make it more permanent (a user can reset this state by signing out)?